azure/README.md

```diff
@@ -1,79 +1,276 @@
+Download kubeconfig locally to access the cluster using `kubectl`:
 
 ```bash
-VM_NAME="REPLACE_ME"
-PEER_POD_NAME="OUTPUT_FROM_ABOVE"
+az aks get-credentials \
+    --resource-group "${RESOURCE_GROUP}" \
+    --name "${CLUSTER_NAME}"
 ```
 
-# Running cloud-api-adaptor
+Label the nodes so that CAA can be deployed on it:
 
-- If using Calico CNI, [configure](https://projectcalico.docs.tigera.io/networking/vxlan-ipip#configure-vxlan-encapsulation-for-all-inter-workload-traffic) VXLAN encapsulation for all inter workload traffic.
+```bash
+kubectl label nodes --all node-role.kubernetes.io/worker=
+```
+
+## Deploy Cloud API Adaptor
+
+> **NOTE**: If you are using Calico CNI on a different Kubernetes cluster, then, [configure](https://projectcalico.docs.tigera.io/networking/vxlan-ipip#configure-vxlan-encapsulation-for-all-inter-workload-traffic) VXLAN encapsulation for all inter workload traffic.
 
-- Create Service Principal for the CAA
+### AKS Resource Group permissions
+
+AKS deploys the actual resources like the worker nodes in another resource group that is named in the format `MC_<aks-resource-group-name>_<cluster-name>_<location>`. For the CAA to be able to create pod VM in the same subnet as the worker nodes of the AKS cluster, run the following command:
 
 ```bash
+export AKS_RG="MC_${RESOURCE_GROUP}_${CLUSTER_NAME}_${LOCATION}"
 az ad sp create-for-rbac \
-  -n peer-pod-vm-creator \
+  -n "caa-${RESOURCE_GROUP}" \
   --role Contributor \
-  --scopes "/subscriptions/$SUBSCRIPTION_ID/resourceGroups/$RESOURCE_GROUP" \
-  --query "{ clientid: appId, secret: password, tenantid: tenant }"
+  --scopes "/subscriptions/${SUBSCRIPTION_ID}/resourceGroups/${AKS_RG}" \
+  --query "{ AZURE_CLIENT_ID: appId, AZURE_CLIENT_SECRET: password }"
+```
+
+From the output of the above command populate the environment variables below:
+
+```bash
+export AZURE_CLIENT_ID="REPLACE_ME"
+export AZURE_CLIENT_SECRET="REPLACE_ME"
+```
+
+### AKS Subnet ID
+
+Fetch the VNET name of that AKS created automatically:
+
+```bash
+export VNET_NAME=$(az network vnet list \
+  --resource-group "${AKS_RG}" \
+  --query "[0].name" \
+  --output tsv)
+```
+
+Export the subnet ID to be used for CAA daemonset deployment:
+
+```bash
+export SUBNET_ID=$(az network vnet subnet list \
+  --resource-group "${AKS_RG}" \
+  --vnet-name "${VNET_NAME}" \
+  --query "[0].id" \
+  --output tsv)
+```
+
+### Populate the `kustomization.yaml` File
+
+Replace the values as needed for the following environment variables:
+
+```bash
+# For regular VMs use something like: Standard_D2as_v5, for CVMs use something like Standard_DC2as_v5.
+export AZURE_INSTANCE_SIZE="REPLACE_ME"
+```
+
+Run the following command to update the [`kustomization.yaml`](../install/overlays/azure/kustomization.yaml) file:
+
+```bash
+export AZURE_SUBSCRIPTION_ID="${SUBSCRIPTION_ID}"
+export AZURE_TENANT_ID="${TENANT_ID}"
+
+sed -i \
+  -e "s|newName: quay.io.*|newName: ${registry}/cloud-api-adaptor|g" \
```

Comment:
I am not sure what the bigger risk is? Missing out on the defaults or installing additional tool. I would take a risk of adding yet another tool. Let me update the instructions to use `yq`.

Reply:
Using `yq` turned out to be ugly quickly. It is because we use env vars exposed like this: 

```yaml
configMapGenerator:
- name: peer-pods-cm
  namespace: confidential-containers-system
  literals:
  - CLOUD_PROVIDER="azure"
  - AZURE_SUBSCRIPTION_ID="" #set
  - AZURE_REGION="eastus" #set
  - AZURE_INSTANCE_SIZE="Standard_D8as_v5" #set
  - AZURE_RESOURCE_GROUP="" #set
```

Now when using yq I would have to do something like this `ya -i '.configMapGenerator[0].literals[1] = "AZURE_SUBSCRIPTION_ID=strenv(AZURE_SUBSCRIPTION_ID)"'`. Now if the value gets upside down it all messes up. So using a standard file!

Reply:
Updated, PTAL.